src/event_loop.rs

```diff
@@ -165,24 +168,33 @@ impl<Io> EventLoop<Io>
 
     // Drain the channel
     //
-    // Returns true if items were received
-    fn drain_recv_channel(&self, state: &mut State) -> bool {
+    // Returns `Ok` if the `EventLoop` should continue running.
+    // `Ok(true)`is returned if items were received
+    //
+    // An `Err` indicates that the event loop should shut down
+    fn drain_recv_channel(&self, state: &mut State) -> Result<bool, ()> {
```

Comment:
I agree that the return type here is awkward. It might be nice to return a new enum type indicating the result. Maybe something like

```rust
enum DrainResult {
    ReceivedItem,
    Empty,
    Shutdown
}
```

would be clearer. Methods can always be implemented on `DrainResult` for convenience. Since you're checking for `is_err` in the current implementation, maybe a method like `is_shutdown` would be useful.